internal/testutil/pkgbuilder/builder.go

```diff
@@ -302,6 +302,7 @@ func (sp *SubPkg) WithSubPackages(ps ...*SubPkg) *SubPkg {
 type Kptfile struct {
 	Upstream     *Upstream
 	UpstreamLock *UpstreamLock
+	Pipeline     *Pipeline
```

Comment:
This testing utility seems to reinvent the wheel. Why can't it just use the actual Kptfile type instead of creating its own and using templating for generation?

Reply:
It is not identical to a Kptfile as it allows specifying repositories and git commit hashes as references that are resolved at test execution time, since we don't know them before we have created the repos and made the commits. As the rest of the pkgbuilder, it also allows function chaining to make it easier to create packages for testing with as little code as possible. 

I have been thinking about replacing the templating. It was convenient what the Kptfile supported setters, but less important now. I just haven't gotten around to it yet.